version/check.go

```diff
@@ -0,0 +1,64 @@
+package version
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+)
+
+const versionTimeout = time.Second * 1
+
+// CheckLatestVersion prints the latest version if the current version is less than latest version
+func CheckLatestVersion(currVersion string, githubReleaseURL string) error {
+	gitClient := http.Client{
+		Timeout: versionTimeout,
+	}
+
+	req, err := http.NewRequest(http.MethodGet, githubReleaseURL, nil)
```

Comment:
If this is specific to github anyway. We can avoid taking full URL and only take `org/repo`

Reply:
Had the same thought but noticed that github gives the API url as a variable, `GITHUB_API_URL` , in [github workflow](https://docs.github.com/en/actions/reference/environment-variables#:~:text=github_api_url).

So was thinking we should embed these variables while building in the applications CI.

In the application then we can just format it and give to the `version.CheckLatest` -
`${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}/releases/latest`

Reply:
Should we change the URL format with this, or continue with the previous one ?
`githubReleaseURL := fmt.Sprintf("https://api.github.com/repos/%s/releases/latest",repoName)`

Reply:
Lets use just repo name as @ravisuhag suggested. Don't think github api url is going to change anytime soon.